staging/src/k8s.io/apiserver/pkg/util/flowcontrol/rule.go

```diff
@@ -0,0 +1,100 @@
+/*
+Copyright 2019 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package flowcontrol
+
+import (
+	rmtypesv1alpha1 "k8s.io/api/flowcontrol/v1alpha1"
+	"k8s.io/apiserver/pkg/authentication/serviceaccount"
+)
+
+func matchesFlowSchema(digest RequestDigest, flowSchema *rmtypesv1alpha1.FlowSchema) bool {
+	for _, policyRule := range flowSchema.Spec.Rules {
+		subjectMatches := false
+		for _, subject := range policyRule.Subjects {
+			if matchesSubject(digest, subject) {
+				subjectMatches = true
+				break
+			}
+		}
+		if !subjectMatches {
+			continue
+		}
+
+		if matchesPolicyRule(digest, policyRule.Rule) {
+			return true
+		}
+	}
+	return false
+}
+
+func matchesSubject(digest RequestDigest, subject rmtypesv1alpha1.Subject) bool {
+	user := digest.User
+	switch subject.Kind {
+	case rmtypesv1alpha1.UserKind:
+		return user.GetName() == subject.Name
+	case rmtypesv1alpha1.GroupKind:
+		return containsString(subject.APIGroup, user.GetGroups(), "")
```

Comment:
The string to seek in this case is `subject.Name`, not `subject.APIGroup`.

Reply:
fixed